server/common/utils/type_conversion_utils.py

```diff
@@ -102,3 +129,10 @@ def can_cast_to_int32(dtype, array_values=None):
                 array_values.min() >= ii32.min and array_values.max() <= ii32.max) or array_values.empty:
```

Comment:
So this gets called a little earlier than it used to, so now it can get a pd.Series with int categories that is not _ordered_, causing min to raise `TypeError`.

```
>>> s = pd.Series([1, 2, 3, 4, 1, 2, 3], dtype="category")
>>> s.dtype.categories
Int64Index([1, 2, 3, 4], dtype='int64')
>>> s.dtype.categories.dtype.kind
'i'
>>> s.min()
...many line traceback...
TypeError: Categorical is not ordered for operation min
you can use .as_ordered() to change the Categorical to an ordered one
```

I believe you want to check if min raises TypeError and if so, `array_values.cat.as_ordered().min()`